src/main/kotlin/lotto/domain/WinningResult.kt

```diff
@@ -1,26 +1,17 @@
 package lotto.domain
 
-class WinningResult(private val winningStatistics: Map<Int, Int>) {
+class WinningResult(private val winningStatistics: Map<Rank, Int>) {
     fun calculateTotalPrize(): Int {
-        return winningStatistics.entries.sumOf { (matchCount, count) ->
-            calculatePrizeForMatch(matchCount, count)
+        return winningStatistics.entries.sumOf { (rank, count) ->
+            rank.prize * count
```

Comment:
[이전 코멘트](https://github.com/next-step/kotlin-lotto/pull/1102#discussion_r1864099891)로 말씀드렸던 부분인데, 
[rank 객체에게 메지시를 보내](https://tecoble.techcourse.co.kr/post/2020-04-28-ask-instead-of-getter/)보면 어떨까요? 
 
```suggestion
            rank.xxxx(count)
```

Reply:
아 ㅎㅎ 사실 이부분 calculatePrize라는 금액 계산 함수를 rank에 넣어 놨었는데,
생각해보니 Rank가 가격 계산을 담당하는 형태가 되서 간단한 계산은 외부에서 보는게 더 직관적일것 같아서 빼놨습니다!
해당 메소드의 이름이 calculateTotalPrize이라 현재에는 보기에도 좀 더 명확해보여서,
좀 더 복잡한 함수가 들어가면 객체 책임 분리하면서 빼놓을까 생각했는데, 어떻게 생각하실까요~~?

Reply:
Rank같은 enum 객체는 팩토리처럼 해당 enum의 값을 생성/선택하는 책임을 주로 가지고, 단순한 데이터 제공은 외부에 노출하는 게 더 자연스러울 수 있다고 생각이 들어서요!

Reply:
질문 주신부분에대한 제 의견을 남겨보자면

 전 enum객체는 팩토리처럼 해당 객체의 값을 생성/선택 하는 책임을 주로 가지고, 비즈니스에 대한 계산은 해당 클래스에서 하기 애매하는 생각이 들어서요!

전 enum 객체로 단순히 하나의 객체라고 생각해요 : )
자신의 역할과 책임을 가지고 다른 객체와 협력하는 하나의 객체라고 생각합니다.

비즈니스에 대한 계산은 해당 클래스에서하기 애매하다는 의견에는 조금 반대되는 생각을 하고있습니다. :thinking_face: